nft/ethereum_payment_obligation.go

```diff
@@ -145,7 +263,21 @@ func (s *ethereumPaymentObligation) MintNFT(ctx context.Context, documentID []by
 		return nil, nil, err
 	}
 
-	registry := common.HexToAddress(registryAddress)
+	// if the role_proof is given check if the role exists with account id as collaborator
+	// at the moment, since we generate role key randomly, this will not work
```

Comment:
I would remove this comment. It will be outdated pretty soon. And the proof itself still works, it's just not the most helpful :)